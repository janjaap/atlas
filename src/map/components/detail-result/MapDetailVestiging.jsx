import React from 'react';
import PropTypes from 'prop-types';

import MapDetailAddressItem from './MapDetailAddressItem';
import MapDetailResultItem from './MapDetailResultItem';
import MapDetailResultStatusItem from './MapDetailResultStatusItem';
import MapDetailResultWrapper from './MapDetailResultWrapper';
import MapDetailVestigingActiviteitenItem from './MapDetailVestigingActiviteitenItem';
import Notification from '../../../shared/components/notification/Notification';

const MapDetailVestiging = ({ panoUrl, vestiging, onMaximize, onPanoPreviewClick }) => {
  const special = vestiging.bijzondereRechtstoestand &&
    (vestiging.bijzondereRechtstoestand.faillissement ||
      vestiging.bijzondereRechtstoestand.surseanceVanBetaling);
  const specialLabel = vestiging.bijzondereRechtstoestand &&
    vestiging.bijzondereRechtstoestand.faillissement ? 'Faillissement' : 'Surseance van betaling';

  return (
    <MapDetailResultWrapper
      panoUrl={panoUrl}
      onMaximize={onMaximize}
      onPanoPreviewClick={onPanoPreviewClick}
      subTitle={vestiging.label}
      title="Vestiging"
    >
      {vestiging.label ? (
        <ul className="map-detail-result__list">
          {special && <li className="map-detail-result__notification">
            <Notification
              level="alert"
              canClose={false}
            >{specialLabel}</Notification>
          </li>}
          <MapDetailResultItem
            label="KvK-nummer"
            value={vestiging.kvkNumber}
          />
          <MapDetailResultItem
            label="Vestigingsnummer"
            value={vestiging.vestigingsnummer}
          />
          <MapDetailAddressItem
            label="Bezoekadres"
            values={vestiging.visitingAddress}
          />
          <MapDetailVestigingActiviteitenItem activities={vestiging.activities} />
          <MapDetailResultItem
            label="Type vestiging"
            value={vestiging.hoofdvestiging ? 'Hoofdvestiging' : 'Nevenvestiging'}
          />
          {special && (<MapDetailResultStatusItem
            label="Soort bijzondere rechtstoestand"
            value={specialLabel}
            status="alert"
          />)}
        </ul>
      ) : (
        <Notification>
          Medewerkers/ketenpartners van Gemeente Amsterdam kunnen inloggen om
          maatschappelijke activiteiten en vestigingen te bekijken.
        </Notification>
      )}
    </MapDetailResultWrapper>
  );
};

MapDetailVestiging.propTypes = {
  panoUrl: PropTypes.string.isRequired,
  vestiging: PropTypes.shape({
    activities: PropTypes.array,
    bijzondereRechtstoestand: PropTypes.shape({
      faillissement: PropTypes.boolean,
      status: PropTypes.string,
      surseanceVanBetaling: PropTypes.boolean
    }),
    kvkNumber: PropTypes.string,
    label: PropTypes.string,
    visitingAddress: PropTypes.object,
    hoofdvestiging: PropTypes.bool,
    vestigingsnummer: PropTypes.string
  }).isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailVestiging;
