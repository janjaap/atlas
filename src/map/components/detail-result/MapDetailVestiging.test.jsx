import React from 'react';
import { shallow } from 'enzyme';

import MapDetailVestiging from './MapDetailVestiging';

describe('MapDetailVestiging', () => {
  it('should render everything', () => {
    const vestiging = {
      activities: [{
        sbiCode: '01',
        sbiDescription: 'SBI Description 1'
      }, {
        sbiCode: '02',
        sbiDescription: 'SBI Description 2'
      }],
      bijzondereRechtstoestand: {
        faillissement: true,
        status: 'Bijzondere rechtstoestand',
        surseanceVanBetaling: false
      },
      kvkNumber: '123456',
      label: 'Vestiging label',
      visitingAddress: { plaats: 'Amsterdam' }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailVestiging
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        vestiging={vestiging}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render `bijzondere rechtstoestand` when `surseance van betaling`', () => {
    const vestiging = {
      activities: [{
        sbiCode: '01',
        sbiDescription: 'SBI Description 1'
      }, {
        sbiCode: '02',
        sbiDescription: 'SBI Description 2'
      }],
      bijzondereRechtstoestand: {
        faillissement: false,
        status: 'Bijzondere rechtstoestand',
        surseanceVanBetaling: true
      },
      kvkNumber: '123456',
      label: 'Vestiging label',
      visitingAddress: { plaats: 'Amsterdam' }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailVestiging
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        vestiging={vestiging}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render `bijzondere rechtstoestand`', () => {
    const vestiging = {
      activities: [{
        sbiCode: '01',
        sbiDescription: 'SBI Description 1'
      }, {
        sbiCode: '02',
        sbiDescription: 'SBI Description 2'
      }],
      bijzondereRechtstoestand: {
        faillissement: false,
        status: 'Bijzondere rechtstoestand',
        surseanceVanBetaling: false
      },
      kvkNumber: '123456',
      label: 'Vestiging label',
      visitingAddress: { plaats: 'Amsterdam' }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailVestiging
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        vestiging={vestiging}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without label', () => {
    const vestiging = {
      activities: [{
        sbiCode: '01',
        sbiDescription: 'SBI Description 1'
      }, {
        sbiCode: '02',
        sbiDescription: 'SBI Description 2'
      }],
      bijzondereRechtstoestand: { status: 'Bijzondere rechtstoestand' },
      kvkNumber: '123456',
      visitingAddress: { plaats: 'Amsterdam' }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailVestiging
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        vestiging={vestiging}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render everything with hoofdvestiging = false', () => {
    const vestiging = {
      activities: [{
        sbiCode: '01',
        sbiDescription: 'SBI Description 1'
      }, {
        sbiCode: '02',
        sbiDescription: 'SBI Description 2'
      }],
      bijzondereRechtstoestand: { status: 'Bijzondere rechtstoestand' },
      hoofdvestiging: false,
      vestigingsnummer: '0001234',
      kvkNumber: '123456',
      label: 'Vestiging label',
      visitingAddress: { plaats: 'Amsterdam' }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailVestiging
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        vestiging={vestiging}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
