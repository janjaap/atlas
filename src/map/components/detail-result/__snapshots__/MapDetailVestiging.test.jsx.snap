// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailVestiging should not render \`bijzondere rechtstoestand\` 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Vestiging label"
  title="Vestiging"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="KvK-nummer"
      value="123456"
    />
    <MapDetailResultItem
      label="Vestigingsnummer"
      value=""
    />
    <MapDetailAddressItem
      label="Bezoekadres"
      values={
        Object {
          "plaats": "Amsterdam",
        }
      }
    />
    <MapDetailVestigingActiviteitenItem
      activities={
        Array [
          Object {
            "sbiCode": "01",
            "sbiDescription": "SBI Description 1",
          },
          Object {
            "sbiCode": "02",
            "sbiDescription": "SBI Description 2",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Type vestiging"
      value="Nevenvestiging"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailVestiging should render \`bijzondere rechtstoestand\` when \`surseance van betaling\` 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Vestiging label"
  title="Vestiging"
>
  <ul
    className="map-detail-result__list"
  >
    <li
      className="map-detail-result__notification"
    >
      <Notification
        canClose={false}
        className=""
        level="alert"
      >
        Surseance van betaling
      </Notification>
    </li>
    <MapDetailResultItem
      label="KvK-nummer"
      value="123456"
    />
    <MapDetailResultItem
      label="Vestigingsnummer"
      value=""
    />
    <MapDetailAddressItem
      label="Bezoekadres"
      values={
        Object {
          "plaats": "Amsterdam",
        }
      }
    />
    <MapDetailVestigingActiviteitenItem
      activities={
        Array [
          Object {
            "sbiCode": "01",
            "sbiDescription": "SBI Description 1",
          },
          Object {
            "sbiCode": "02",
            "sbiDescription": "SBI Description 2",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Type vestiging"
      value="Nevenvestiging"
    />
    <MapDetailResultStatusItem
      label="Soort bijzondere rechtstoestand"
      status="alert"
      value="Surseance van betaling"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailVestiging should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Vestiging label"
  title="Vestiging"
>
  <ul
    className="map-detail-result__list"
  >
    <li
      className="map-detail-result__notification"
    >
      <Notification
        canClose={false}
        className=""
        level="alert"
      >
        Faillissement
      </Notification>
    </li>
    <MapDetailResultItem
      label="KvK-nummer"
      value="123456"
    />
    <MapDetailResultItem
      label="Vestigingsnummer"
      value=""
    />
    <MapDetailAddressItem
      label="Bezoekadres"
      values={
        Object {
          "plaats": "Amsterdam",
        }
      }
    />
    <MapDetailVestigingActiviteitenItem
      activities={
        Array [
          Object {
            "sbiCode": "01",
            "sbiDescription": "SBI Description 1",
          },
          Object {
            "sbiCode": "02",
            "sbiDescription": "SBI Description 2",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Type vestiging"
      value="Nevenvestiging"
    />
    <MapDetailResultStatusItem
      label="Soort bijzondere rechtstoestand"
      status="alert"
      value="Faillissement"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailVestiging should render everything with hoofdvestiging = false 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Vestiging label"
  title="Vestiging"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="KvK-nummer"
      value="123456"
    />
    <MapDetailResultItem
      label="Vestigingsnummer"
      value="0001234"
    />
    <MapDetailAddressItem
      label="Bezoekadres"
      values={
        Object {
          "plaats": "Amsterdam",
        }
      }
    />
    <MapDetailVestigingActiviteitenItem
      activities={
        Array [
          Object {
            "sbiCode": "01",
            "sbiDescription": "SBI Description 1",
          },
          Object {
            "sbiCode": "02",
            "sbiDescription": "SBI Description 2",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Type vestiging"
      value="Nevenvestiging"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailVestiging should render without label 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle=""
  title="Vestiging"
>
  <Notification
    canClose={true}
    className=""
    level="info"
  >
    Medewerkers/ketenpartners van Gemeente Amsterdam kunnen inloggen om maatschappelijke activiteiten en vestigingen te bekijken.
  </Notification>
</MapDetailResultWrapper>
`;
